src/main/java/mate/jdbc/dao/impl/CarDaoImpl.java

```diff
@@ -0,0 +1,229 @@
+    }
+
+    @Override
+    public Car update(Car car) {
+        String query = "UPDATE cars SET model = ?, manufacturer_id = ?"
+                + " WHERE id = ? AND is_deleted = FALSE;";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement updateCarStatement =
+                        connection.prepareStatement(query)) {
+            updateCarStatement.setString(1, car.getModel());
+            updateCarStatement.setLong(2, car.getManufacturer().getId());
+            updateCarStatement.setLong(3, car.getId());
+            updateCarStatement.executeUpdate();
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can`t update a car: " + car, e);
+        }
+        removeDriversFromCar(car.getId());
+        insertDriversToCar(car);
+        return car;
+    }
+
+    @Override
+    public boolean delete(Long id) {
+        String query = "UPDATE cars SET is_deleted = TRUE WHERE id = ?;";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement deleteCarStatement =
+                        connection.prepareStatement(query)) {
+            deleteCarStatement.setLong(1, id);
+            return deleteCarStatement.executeUpdate() > 0;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't delete a car by id " + id, e);
+        }
+    }
+
+    @Override
+    public List<Car> getAllByDriver(Long driverId) {
+        String query = "SELECT c.id AS car_id, c.model AS car_model,"
+                + " m.id AS manufacturer_id, m.name AS manufacturer_name,"
+                + " m.country AS manufacturer_country"
+                + " FROM cars c"
+                + " INNER JOIN manufacturers m"
+                + " ON c.manufacturer_id = m.id"
+                + " INNER JOIN cars_drivers cd"
+                + " ON c.id = cd.car_id"
+                + " WHERE driver_id = ?;";
+        List<Car> cars = new ArrayList<>();
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement getAllCarByDriverStatement =
+                        connection.prepareStatement(query)) {
+            getAllCarByDriverStatement.setLong(1, driverId);
+            ResultSet resultSet = getAllCarByDriverStatement.executeQuery();
+            while (resultSet.next()) {
+                cars.add(parseCar(resultSet));
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can`t get all cars by driver with id: "
+                    + driverId, e);
+        }
+        if (!cars.isEmpty()) {
+            for (Car car : cars) {
+                car.setDrivers(getDriversList(car.getId()));
+            }
+        }
+        return cars;
+    }
+
+    private Car parseCar(ResultSet resultSet) throws SQLException {
+        Manufacturer manufacturer = new Manufacturer();
+        manufacturer.setId(resultSet.getObject(1, Long.class));
```

Comment:
While setting the id of the manufacturer, you are using getObject(1, Long.class) which will retrieve the car's id instead of the manufacturer's id. You should use resultSet.getObject('manufacturer_id', Long.class) instead to get the correct id.